pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,500 @@
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors import TabularCPD
+        >>> student = DBN('D','G','I','S','L')
+        >>> grade_cpd = TabularCPD('G', 3, [[0.3, 0.05, 0.9, 0.5],[0.4, 0.25, 0.08, 0.3],[0.3, 0.7, 0.02, 0.2]], ['I','D'],[2,2])
+        >>> student.add_dbn_cpds(0, grade_cpd)
+        >>> student.get_dbn_cpds('G',0)
+        ╒═════╤═════╤══════╤══════╤═════╕
+        │ D   │ D_0 │ D_0  │ D_1  │ D_1 │
+        ├─────┼─────┼──────┼──────┼─────┤
+        │ I   │ I_0 │ I_1  │ I_0  │ I_1 │
+        ├─────┼─────┼──────┼──────┼─────┤
+        │ G_0 │ 0.3 │ 0.05 │ 0.9  │ 0.5 │
+        ├─────┼─────┼──────┼──────┼─────┤
+        │ G_1 │ 0.4 │ 0.25 │ 0.08 │ 0.3 │
+        ├─────┼─────┼──────┼──────┼─────┤
+        │ G_2 │ 0.3 │ 0.7  │ 0.02 │ 0.2 │
+        ╘═════╧═════╧══════╧══════╧═════╛
+        """
+        if node:
+            if timeslice:
+                if (node,timeslice) in self.dbn_cpds:
+                    return self.dbn_cpds[(node, timeslice)]
+            else:
+                list_of_cpds = []
+                for time_state in range(self.timestate + 1):
+                    list_of_cpds.append(self.dbn_cpds[(node, time_state)])
+                return list_of_cpds
+        else:
+            return self.dbn_cpds
+
+    def compute_initial_state(self):
+        """
+        This method will automatically re-adjust the cpds and the edges added to the bayesian network.
+        If an edge that is added as an intra time slice edge in the 0th timeslice, this method will automatically
+        add it in the 1st timeslice. It will also add the cpds.
+        However, to call this method, one needs to add cpds as well as the edges in the
+        bayesian network of the whole skeleton including the 0th and the 1st timeslice,.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors import TabularCPD
+        >>> student = DBN()
+        >>> student.add_nodes_from(['D','G','I','S','L'])
+        >>> student.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1))])
+        >>> grade_cpd = TabularCPD('G', 3, [[0.3, 0.05, 0.9, 0.5],[0.4, 0.25, 0.08, 0.3],[0.3, 0.7, 0.02, 0.2]], ['I','D'],[2,2])
+        >>> intel_cpd = TabularCPD('I', 2, [[0.9, 0.4],[0.1, 0.6]],['I'],[2])
+        >>> diff_cpd = TabularCPD('D', 2, [[0.3, 0.2],[0.7, 0.8]], ['D'],[2])
+        >>> d_cpd = TabularCPD('D', 2, [[0.6, 0.4]])
+        >>> i_cpd = TabularCPD('I', 2, [[0.7, 0.3]])
+        >>> student.add_dbn_cpds(0, grade_cpd, d_cpd, i_cpd)
+        >>> student.add_dbn_cpds(1, diff_cpd, intel_cpd)
+        >>> student.compute_initial_state()
+        """
+
+        if not self.computed:
+            for node in self.dbn_cpds:
+                list_of_parents = super().get_parents(node)
+                if all(x[1] == list_of_parents[0][1] for x in list_of_parents) and list_of_parents != [] and (
```

Comment:
Could you please add a line of comment above this check stating what are these checks. This makes the code more readable and easy to understand for a newcomer.